compiler/rustc_infer/src/infer/glb.rs

```diff
@@ -155,4 +155,9 @@ impl<'tcx> ObligationEmittingRelation<'tcx> for Glb<'_, '_, 'tcx> {
     fn register_obligations(&mut self, obligations: PredicateObligations<'tcx>) {
         self.fields.register_obligations(obligations);
     }
+
+    fn alias_relate_direction(&self) -> ty::AliasRelationDirection {
+        // FIXME(lazy_norm): This isn't right, I think?
```

Comment:
I think this + the `lub.rs` FIXME are not `lazy_norm` but actually `deferred_projection_equality`. Even without lazy norm we can end up with unnormalizable projections that might later be normalizable. 

You could imagine I think a match statement with two arms, both returning projections that cant yet be normalized i.e. `<?0 as Trait<'a>>::Assoc` and the other `<?1 as Trait<'b>>::Assoc` which later `?0` and `?1` get inferred allowing them to be normalized to `&'a ()` and `&'b ()` respectively. We'd want the type of the match statement to be some third lifetime `?'c` that is a supertype of both. Wheras using `AliasRelationDirection::Equate` would require `'a == 'b`.

Reply:
This FIXME should probably block `-Ztrait-solver=next` being stabilized too although I think at this point `deferred_projection_equality` FIXME's are part of `-Ztrait-solver=next` work